experimental/Schemes/Types.jl

```diff
@@ -1333,3 +1334,153 @@ end
                typeof(R)}(R)
   end
 end
+
+########################################################################
+# Ideal sheaves on covered schemes                                     #
+########################################################################
+@Markdown.doc """
+    IdealSheaf <: AbsPreSheaf
+
+A sheaf of ideals on an `AbsCoveredScheme` ``X``.
+
+**note:** The admissible open sets for these sheaves are restricted 
+to instances of `AbsSpec` since we only have types for ideals the 
+coordinate rings of the latter.
+"""
+@attributes mutable struct IdealSheaf{SpaceType, OpenType, OutputType,
+                                      RestrictionType, ProductionFuncType,
+                                      RestrictionFuncType,
+                                      PreSheafType
+                                     } <: AbsPreSheaf{
+                                                      SpaceType, OpenType, 
+                                                      OutputType, RestrictionType
+                                                     }
+  ID::IdDict{AbsSpec, Ideal} # the ideals on the basic patches of the default covering
+  OOX::StructureSheafOfRings # the structure sheaf on X
+  I::PreSheafType # the underlying presheaf of ideals for caching
+
+  ### Ideal sheaves on covered schemes
+  function IdealSheaf(X::AbsCoveredScheme, ID::IdDict{AbsSpec, Ideal}; 
+      check::Bool=true
+    )
+    OOX = StructureSheafOfRings(X)
+
+    ### Checks for open containment. 
+    #
+    # We allow the following cases:
+    #
+    #  * U::PrincipalOpenSubset in W===ambient_scheme(U) in the basic charts of X
+    #  * U::PrincipalOpenSubset ⊂ V::PrincipalOpenSubset with ambient_scheme(U) === ambient_scheme(V) in the basic charts of X
+    #  * U::PrincipalOpenSubset ⊂ V::PrincipalOpenSubset with ambient_scheme(U) != ambient_scheme(V) both in the basic charts of X
+    #    and U and V contained in the glueing domains of their ambient schemes
+    #  * U::AbsSpec ⊂ U::AbsSpec in the basic charts of X
+    #  * U::AbsSpec ⊂ X for U in the basic charts
+    #  * U::PrincipalOpenSubset ⊂ X with ambient_scheme(U) in the basic charts of X
```

Comment:
Looks like this should be part of the documentation in some (simplified) form. since it is relevant for the user.

Reply:
Yes. I was wondering how to do the documentation for the sheaves. I felt that it would be good somehow to talk about the `AbsPreSheaf` interface first and not repeat everything for every sheaf again and again. But I didn't solve this, yet. I'll do it.